packages/fxa-settings/src/components/ConnectedServices/index.tsx

```diff
@@ -55,9 +61,136 @@ export const ConnectedServices = () => {
+  const viewName = 'settings.clients.disconnect';
+
+  // The Confirm Disconnect modal is shown when a user clicks 'Sign Out' on a sync service.
+  // It asks the user to confirm they want to disconnect, and answer a survey question explaining
+  // why they are disconnecting.
+  const [
+    confirmDisconnectModalRevealed,
+    revealConfirmDisconnectModal,
+    hideConfirmDisconnectModal,
+  ] = useBooleanState();
+  const [disconnectingClient, setDisconnectingClient] = useState<
+    AttachedClient
+  >();
+  const [disconnectReason, setDisconnectReason] = useState<string>();
+  const [submitStartTime, setSubmitStartTime] = useState<number>();
+
+  // After the user confirms they want to disconnect from sync in Confirm Disconnect modal,
+  // if their reason was a lost/stolen device, or a suspicious device, then we show them
+  // an informative modal with some advice on next steps to take.
+  const [
+    adviceModalRevealed,
+    revealAdviceModal,
+    hideAdviceModal,
+  ] = useBooleanState();
+
+  const clearDisconnectingState = (
+    errorMessage?: string,
+    error?: ApolloError | undefined
+  ) => {
+    hideConfirmDisconnectModal();
+    // TODO: if I uncomment these lines, stuff breaks. What's the right way to coordinate
+    // hanging on to enough local state until we're done? Maybe not pass around the AttachedClient,
+    // but instead pass around copies of the data? Like the name, clientId, lastAccessTime, etc?
+    //setDisconnectingClient(undefined);
+    //setDisconnectReason(undefined);
+    //setSubmitStartTime(0);
+    if (errorMessage) {
+      alertBar.error(errorMessage, error);
+    }
+  };
+
+  const onSignOutClick = (client: AttachedClient) => {
+    setDisconnectingClient(client!);
+    revealConfirmDisconnectModal();
+  };
+
+  const onConfirmDisconnect = () => {
+    setSubmitStartTime(Date.now());
+
+    if (!disconnectingClient) {
+      return clearDisconnectingState('Client not found, unable to disconnect');
+    }
+
+    // TODO: does this need to be a _flow_ event?
+    logViewEvent('settings.clients.disconnect', `submit.${disconnectReason}`);
+
+    deleteConnectedService({
+      variables: {
+        input: {
+          clientId: disconnectingClient.clientId,
+          deviceId: disconnectingClient.deviceId,
+          sessionTokenId: disconnectingClient.sessionTokenId,
+          refreshTokenId: disconnectingClient.refreshTokenId,
+        },
+      },
+    });
+  };
+
+  const ATTACHED_CLIENT_DISCONNECT_MUTATION = gql`
+    mutation attachedClientDisconnect($input: AttachedClientDisconnectInput!) {
+      attachedClientDisconnect(input: $input) {
+        clientMutationId
+      }
+    }
+  `;
+
+  const [deleteConnectedService] = useMutation(
+    ATTACHED_CLIENT_DISCONNECT_MUTATION,
+    {
+      onCompleted: () => {
+        logViewEvent(
+          viewName,
+          `timing.clients.disconnect.${Date.now() - submitStartTime!}`
```

Comment:
This logs a `settings.clients.disconnect.timing.clients.disconnect.nnn` event, but we want a `flow.timing.clients.disconnect.nnn` event.  I don't think we've started logging any flow timing events in new settings.